test/extended/apiserver/kubeconfigs.go

```diff
@@ -29,13 +32,25 @@ var _ = g.Describe("[Conformance][sig-api-machinery][Feature:APIServer] local ku
 			framework.Logf("Discovered %d master nodes.", len(masterNodes.Items))
 			o.Expect(masterNodes.Items).NotTo(o.HaveLen(0))
 			for _, master := range masterNodes.Items {
-				g.By("Testing master node " + master.Name)
-				kubeconfigPath := "/etc/kubernetes/static-pod-resources/kube-apiserver-certs/secrets/node-kubeconfigs/" + kubeconfig
-				framework.Logf("Verifying kubeconfig %q on master %s", master.Name)
-				out, err := oc.AsAdmin().Run("debug").Args("node/"+master.Name, "--", "chroot", "/host", "/bin/bash", "-euxo", "pipefail", "-c", fmt.Sprintf(`oc --kubeconfig "%s" get namespace kube-system`, kubeconfigPath)).Output()
+				retry, err := testNode(oc, kubeconfig, master.Name)
+				if retry {
+					g.By("There was a retryable error for " + fmt.Sprintf("%s/%s", master.Name, kubeconfig))
+					_, err = testNode(oc, kubeconfig, master.Name)
+				}
 				o.Expect(err).NotTo(o.HaveOccurred())
-				framework.Logf(out)
 			}
 		})
 	}
 })
+
+func testNode(oc *exutil.CLI, kubeconfig, masterName string) (bool, error) {
+	g.By("Testing master node " + masterName)
+	kubeconfigPath := "/etc/kubernetes/static-pod-resources/kube-apiserver-certs/secrets/node-kubeconfigs/" + kubeconfig
+	framework.Logf("Verifying kubeconfig %q on master %q", kubeconfig, masterName)
+	out, err := oc.AsAdmin().Run("debug").Args("node/"+masterName, "--", "chroot", "/host", "/bin/bash", "-euxo", "pipefail", "-c",
+		fmt.Sprintf(`oc --kubeconfig "%s" get namespace kube-system`, kubeconfigPath)).Output()
+	framework.Logf(out)
+	// retry error when kube-apiserver was temporarily unavailable
+	retry := strings.Contains(out, "The connection to the server localhost:6443 was refused")
```

Comment:
would you be open to adding a precondition before running the test instead? 
not sure this is the only error we might get, for example on an IPv6 cluster we might get `[::1]:6443: connect: connection refused`

we could check if the cluster is in stable condition (not progressing, the pods on the same revision) for X min.

Reply:
Checking progressing might slow this down, but we can add that as last resort, re-tries seems simpler, b/c theoretically we can even pass the test when rollout is in progress.

Reply:
Added check for ipv6, though